### PR TITLE
Feature/save to storage

### PR DESCRIPTION
**Pull request by davidleach724**
# Description
This feature will incorporate saving the idea cards to local storage as well as retrieving them upon page load. 

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?
- [x] open index.html
- [x] console.log()
- [x] dev tools

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Idea.js [hunk 1]
```diff
@@ -1,27 +1,30 @@
...
     this.title = title;
     this.body = body;
-    this.star = false;
-    this.image = 'assets/star.svg';
+    this.id = id || Date.now();
+    this.star = star || false;
+    this.image = image || 'assets/star.svg';
   }
 
   saveToStorage() {
```
Comment (ttarlov) [inline src/Idea.js:10]: Good job on the local store implementation!

File: src/Idea.js [hunk 2]
```diff
@@ -1,8 +1,8 @@
 class Idea {
   constructor(title, body, id, star, image) {
-    this.id = id || Date.now();
     this.title = title;
     this.body = body;
+    this.id = id || Date.now();
```
Comment (williamphelps13) [inline src/Idea.js:5]: Good catch on this. I didn't notice that, but on looking back that's a necessary improvement.

**Comment by williamphelps13** [comment]
I tested the saveToStorage functionality and looks solid! Good job on starting refactoring by adding semi-colons, removing spacing, and fixing syntax.

File: src/main.js [hunk 3]
```diff
@@ -104,15 +105,37 @@ function favoriteIdeaCard(event) {
...
 }
 
+function retrieveFromStorage() {
+  var keys = Object.keys(localStorage);
+
+  for (var i = 0; i < keys.length; i++) {
+    var cardDeets = localStorage.getItem(keys[i]);
+    var cardDeetsObj = JSON.parse(cardDeets);
+    var newCard = new Idea(
+      cardDeetsObj.title,
```
Comment (williamphelps13) [inline src/main.js:128]: Spacing makes it easier to read!

**Review by williamphelps13 — APPROVED** [review]

**Pull request merged**